### PR TITLE
Returning the error 400 instead of 500 when creating a new object with non-existent foreign keys.

### PR DESCRIPTION
**Pull request by estush**
Resolving #7

### CODE REVIEW

**Review by RutiRozenberg — CHANGES_REQUESTED** [review]
Hi @estush 

Great job on the code! I appreciate your effort in implementing the feature. I have some improvements based on our project conventions:

1. Variable Names: Please revert any variable names that you did not create and have changed back to their original names.
2. Underscores: Remove underscores before variables according to our project conventions.
3. Default Variable Permission: Default variable permission is private, no need to add it manually.
4. Validation Function: Create a function to check if linked object keys are linked to existing objects. Use this function during object creation and updates to prevent duplicate code.Your function should be last to avoid conflicts.
5. Data Validation: Ensure data validation is performed during updates as well, not just during creation.

Regarding the file: `WebApi/Controllers/CommentController.cs`
- Rearrange functions to match the original project order. 
- Remove the following lines:
` <<<<<<< HEAD `(line 64)
` =======` (line 171)
` >>>>>>> 9a62b7be341259c3625e9f3811aaf962b0b92384` (line 175)

Please let me know if you have any questions or need further clarification on these points.

**Comment by RutiRozenberg** [comment]
Hi @estush , I noticed that the commit messages could be more descriptive and specific. It would be helpful to use meaningful commit message titles that explain the changes made in each commit. This will improve code readability and make it easier to understand the purpose of each change. Please consider updating the commit messages accordingly. Thank you!

**Comment by estush** [comment]
OK

‫בתאריך יום ה׳, 26 בספט׳ 2024 ב-10:56 מאת ‪Ruti Rozenberg‬‏ <‪
***@***.***‬‏>:‬

**Comment by MichalYiloz** [comment]
#10  
-my Issues
Thank you!

**Comment by DvoraShechter1** [comment]
Conflicts must be dealt with before requesting a review.

**Review by RutiRozenberg — DISMISSED** [review]

File: WebApi/Controllers/CommentController.cs [hunk 1]
```diff
@@ -13,114 +14,129 @@ namespace WebApi.Controllers
...
+
+        public CommentController(ICommentService cService,
+            IDiscussionService dService, ILogger<CommentController> logService, UserValidations usValidate, DiscussionValidations disValidations)
         {
-            CommentService = service;
-            this.logger = logger;
+            commentService = cService;
+            discussionService = dService;
+            logger = logService;
+            userValidations = usValidate; // ����� �� UserValidations
```
Comment (DvoraShechter1) [inline WebApi/Controllers/CommentController.cs]: Please remove unclear comment.

File: WebApi/Controllers/CommentController.cs [hunk 2]
```diff
@@ -159,4 +175,5 @@ public async Task<IActionResult> GetByDiscussionId(int discussionId)
 
     }
 
+
```
Comment (DvoraShechter1) [inline WebApi/Controllers/CommentController.cs]: Please remove an empty line.

**Review by DvoraShechter1 — CHANGES_REQUESTED** [review]
Just a few small changes and this PR will be good to go.

**Review by DvoraShechter1 — APPROVED** [review]
LGTM!

**Pull request merged**